finagle-redis/src/main/scala/com/twitter/finagle/redis/ScriptCommands.scala

```diff
@@ -0,0 +1,147 @@
+package com.twitter.finagle.redis
+
+import com.twitter.finagle.redis.protocol._
+import com.twitter.finagle.redis.util.ReplyFormat
+import com.twitter.util.Future
+import org.jboss.netty.buffer.{ChannelBuffers, ChannelBuffer}
+
+trait Scripts { self: BaseClient =>
+  val handleUnit: PartialFunction[Reply, Future[Unit]] = {
+    case StatusReply(_) | EmptyBulkReply() => Future.Unit
+  }
+
+  val handleLong: PartialFunction[Reply, Future[Long]] = {
+    case IntegerReply(n) => Future.value(n)
+  }
+
+  val handleBool: PartialFunction[Reply, Future[Boolean]] = {
+    case IntegerReply(n)  => Future.value(n == 1)
+    case EmptyBulkReply() => Future.False
+    case StatusReply(_)   => Future.True
+  }
+
+  val handleBuffer: PartialFunction[Reply, Future[ChannelBuffer]] = {
+    case BulkReply(message) => Future.value(message)
+    case EmptyBulkReply() => Future.value(ChannelBuffers.EMPTY_BUFFER)
+  }
+
+  val handleBuffers: PartialFunction[Reply, Future[Seq[ChannelBuffer]]] = {
+    case MBulkReply(messages) => Future.value(ReplyFormat.toChannelBuffers(messages))
+    case EmptyMBulkReply()    => Future.Nil
+  }
+
+  def retryOnNoScript[T](retry: () => T): PartialFunction[Reply, T] = {
+    // NOTE: we cannot make retry call-by-value.
```

Comment:
Do you mean "call-by-name"?

I'm not sure that I follow your reasoning, especially since I think call-by-name is implemented the same way under the hood as what we're doing now.